### PR TITLE
Dynamic configuration service

### PR DESCRIPTION
**Pull request by popzxc**
## Overview

Attempt to adapt static configuration service to the new architecture.

### CODE REVIEW

File: exonum/src/proto/schema/exonum/configuration.proto [hunk 1]
```diff
@@ -47,3 +47,11 @@ message ProposeData {
...
   uint64 num_validators = 3;
 }
+
+// Init data for Configuration service.
+message ConfigurationServiceInit {
+  // Check whether we want to use a custom majority count.
+  bool is_custom_majority_count = 1;
+  // Custom majority count.
+  uint32 majority_count = 2;
+}
```
Comment (aleksuss) [inline exonum/src/proto/schema/exonum/configuration.proto]: new line ? 🙃

File: exonum/src/runtime/configuration_new/config.rs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Config for Configuration service.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ConfigurationServiceConfig {
+    /// Number of votes required to commit the new configuration.
+    /// This value should be greater than 2/3 and less or equal to the
+    /// validators count.
+    pub majority_count: Option<u16>,
```
Comment (alekseysidorov) [inline exonum/src/runtime/configuration_new/config.rs]: What do you think about `quorum` name?

File: exonum/src/runtime/configuration_new/schema.rs [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    /// Returns a `Propose` transaction with a particular configuration hash.
+    pub fn propose(&self, cfg_hash: &Hash) -> Option<Propose> {
+        self.propose_data_by_config_hash()
+            .get(cfg_hash)?
+            .tx_propose
+            .into()
+    }
+
+    /// Returns a list of votes for the proposal corresponding to the given configuration hash.
+    #[cfg_attr(feature = "cargo-clippy", allow(clippy::let_and_return))]
```
Comment (alekseysidorov) [inline exonum/src/runtime/configuration_new/schema.rs]: You can just write `#[allow(clippy::let_and_return)]`

File: exonum/src/runtime/configuration_new/mod.rs [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+use schema::VotingDecision;
+use transactions::{enough_votes_to_commit, VotingContext};
+use config::ConfigurationServiceConfig;
+
+/// Service identifier for the configuration service.
+pub const SERVICE_ID: u16 = 1;
+/// Configuration service name.
+pub const SERVICE_NAME: &str = "configuration";
+
+service_interface! {
```
Comment (alekseysidorov) [inline exonum/src/runtime/configuration_new/mod.rs]: Is it possible to avoid macros?

File: exonum/src/runtime/configuration_new/config.rs (hunk 2 above)
Comment (popzxc) [reply exonum/src/runtime/configuration_new/config.rs]: I think that it makes not much difference from the used one.

File: exonum/src/runtime/configuration_new/mod.rs (hunk 4 above)
Comment (popzxc) [reply exonum/src/runtime/configuration_new/mod.rs]: For now not really.

**Comment by popzxc** [comment]
> As I understand it is just part 1 and it is just old config service.

Exactly. This PR only gives old config service a new interface.

**Pull request merged**